extension/healthcheckextension/config.go

```diff
@@ -33,11 +36,42 @@ type Config struct {
 	// check status.
 	// The default endpoint is "0.0.0.0:13133".
 	TCPAddr confignet.TCPAddr `mapstructure:",squash"`
+
+	// metricsHealthCheck returns exporters functional health check settings
+	MetricsHealthCheck metricsHealthCheckSettings `mapstructure:"metrics_health_check"`
 }
 
 var _ config.Extension = (*Config)(nil)
 
 // Validate checks if the extension configuration is valid
 func (cfg *Config) Validate() error {
+	_, err := time.ParseDuration(cfg.MetricsHealthCheck.Interval)
+	if err != nil {
+		return err
+	}
+	if cfg.TCPAddr.Endpoint == "" {
+		return errors.New("bad config: endpoint must be specified")
+	}
+	if cfg.MetricsHealthCheck.ExporterFailureThreshold <= 0 {
+		return errors.New("bad config: exporter_failure_threshold expects a positive number")
+	}
 	return nil
 }
+
+type metricsHealthCheckSettings struct {
+	// Enabled indicates whether to not enable metrics health check.
+	Enabled bool `mapstructure:"enabled"`
+	// Interval the time range to check healthy status based on the exporter failures
+	Interval string `mapstructure:"interval"`
+	// ExporterFailureThreshold is the threshold of failure numbers status during the Interval
+	ExporterFailureThreshold int `mapstructure:"exporter_failure_threshold"`
+}
+
+// DefaultMetricsHealthCheckSettings returns the default settings for metricsHealthCheck.
+func DefaultMetricsHealthCheckSettings() metricsHealthCheckSettings {
```

Comment:
I believe this can be private

Reply:
Actually we probably don't need a separate method, just add it to the existing `createDefaultConfig` method I think. Or otherwise, let's  move this method to `factory.go` so the defaults are together

Reply:
got it, will move to factory.